codegen/smithy-ruby-codegen-test/model/component-test/main.smithy

```diff
@@ -96,3 +97,62 @@ structure ClientError {
 @error("server")
 @retryable(throttling: true)
 structure ServerError {}
+
+@http(method: "POST", uri: "/default_kitchen_sink")
+operation DefaultKitchenSink {
+    input: DefaultKitchenSinkInputOutput,
+    output: DefaultKitchenSinkInputOutput,
+    errors: [],
+}
+
+structure DefaultKitchenSinkInputOutput {
```

Comment:
Good idea. Do we also test defaults without required? I would think we need to do that, but for at least a few types of members like one simple type (string), one complex type (structure).

Reply:
> Do we also test defaults without required?
Yes, the `parsm_spec` verifies the member (without default + required) doesn't exist for an empty params.

Reply:
What I wanted to check was a member with `@default` trait that is NOT `@required`, and to verify that we DON'T use default values for it.

Reply:
I'm not sure I follow.. can you show me?